crates/miden-tx/src/tests/kernel_tests/test_faucet.rs

```diff
@@ -620,6 +614,51 @@ fn test_burn_non_fungible_asset_fails_inconsistent_faucet_id() {
     assert_execution_error!(process, ERR_FAUCET_NON_FUNGIBLE_ASSET_TO_BURN_NOT_FOUND);
 }
 
+// IS NON FUNGIBLE ASSET ISSUED TESTS
+// ================================================================================================
+
+#[test]
+fn test_is_non_fungible_asset_issued_succeeds() {
+    let tx_context = TransactionContextBuilder::with_non_fungible_faucet(
+        NonFungibleAsset::mock_issuer().into(),
+        ONE,
+        false,
+    )
+    .build();
+
+    let non_fungible_asset_1 = NonFungibleAsset::mock(&NON_FUNGIBLE_ASSET_DATA);
+    let non_fungible_asset_2 = NonFungibleAsset::mock(&NON_FUNGIBLE_ASSET_DATA_2);
+
+    let code = format!(
+        "
+        use.kernel::prologue
+        use.miden::faucet
+
+        begin
+            exec.prologue::prepare_transaction
+
+            # check that NON_FUNGIBLE_ASSET_DATA_2 is already issued
+            push.{non_fungible_asset_2}
+            exec.faucet::is_non_fungible_asset_issued
+
+            # use error code 9998 to assert that asset is issued
+            eq.1 assert.err=9998
```

Comment:
This means that `non_fungible_asset_2` has already been issued, right? How does it get issues? (maybe worth adding a comment somewhere about how this happens).

Reply:
This asset is "issued" during the mock faucet creation, so it is already in the map of issued assets. I'll add a comment to explain this.